### PR TITLE
Drop Ruby 2.2 support

### PR DESCRIPTION
**Pull request by mvz**
## Summary

Drop support for Rubies older than 2.3

## Details

* Bump required Ruby version
* Bump RuboCop target ruby version
* Update RuboCop since latest version supports Ruby 2.3 and up
* Fix new RuboCop offenses
* Remove outdated version checks

## Motivation and Context

This simplifies future maintenance by limiting set of supported Ruby versions. Dropping support for 2.3 before the final release is a distinct possibility. Projects that need support for older Rubies can stick with Aruby 0.x until they're ready.

## How Has This Been Tested?

Travis will tell ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (cleanup of codebase without changing any existing functionality)
- [ ] Update documentation

## Checklist:

- [ ] I've added tests for my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by mvz** [comment]
After this, I think we can release 1.0.0.alpha3.

**Comment by mvz** [comment]
I'm going to need to fix issues due to the frozen string comments.

**Comment by mvz** [comment]
@luke-hill I'm going to release alpha3 first.

File: Gemfile [hunk 1]
```diff
@@ -24,9 +19,7 @@ end
 # Tools to run during development
 group :development do
   # Needed for lint:yard:junk task
-  if RUBY_VERSION >= '2.3.0'
-    gem 'yard-junk', '~> 0'
-  end
+  gem 'yard-junk', '~> 0'
```
Comment (luke-hill) [inline Gemfile]: can we look into this, this puts no restrictions on it. (A bug with the way gemfiles work)

**Comment by luke-hill** [comment]
I've made the minor fixes @mvz just to try help you out. If CI passes I'll merge it in and crack on with some more auto-tidy stuff in the `master` branch.

**Comment by mvz** [comment]
Thanks @luke-hill. Merge at will! :smile:

**Pull request merged**